weave/trace_server/image_completion.py

```diff
@@ -0,0 +1,215 @@
+    project_id: Optional[str] = None,
+    wb_user_id: Optional[str] = None,
+) -> tsi.ImageGenerationCreateRes:
+    """Generate images using LiteLLM image generation.
+
+    Args:
+        api_key (Optional[str]): The API key for the LLM provider.
+        inputs (dict[str, Any]): The input parameters for image generation.
+        provider (Optional[str]): The provider name.
+        base_url (Optional[str]): Custom base URL for the API.
+        extra_headers (Optional[dict[str, str]]): Additional headers.
+        trace_server (Optional[Any]): The trace server instance for file storage.
+        project_id (Optional[str]): The project ID for file storage.
+        wb_user_id (Optional[str]): The user ID for object creation.
+
+    Returns:
+        tsi.ImageGenerationCreateRes: The image generation response.
+
+    Examples:
+        Generate an image with a prompt::
+
+            response = lite_llm_image_generation(
+                api_key="sk-...",
+                inputs={"model": "dall-e-3", "prompt": "A cat wearing a hat"}
+            )
+    """
+    from litellm import image_generation
+
+    model_name = inputs.get("model", "")
+
+    if not model_name:
+        raise InvalidRequest("Model name is required")
+
+    try:
+        # Make the image generation API call
+        # Pass all parameters so LiteLLM can create proper model identifier for cost calculation
+        image_params = {
+            "model": inputs.get("model"),
+            "prompt": inputs.get("prompt"),
+            "api_key": api_key,
+        }
+
+        # Add image generation parameters based on model capabilities
+
+        # Handle 'n' parameter (number of images)
+        # DALL-E 3 only supports n=1, others can support multiple
+        if model_name == "dall-e-3":
+            image_params["n"] = 1  # DALL-E 3 only supports 1 image
+        else:
+            image_params["n"] = inputs.get("n", 1)
+
+        # Only set quality parameter for models that support it (DALL-E 3 and gpt-image-1)
+        if model_name in ["dall-e-3", "gpt-image-1"]:
+            default_quality = "high" if model_name == "gpt-image-1" else "standard"
+            image_params["quality"] = default_quality
+
+        # Only set style parameter for DALL-E 3
+        if model_name == "dall-e-3":
+            image_params["style"] = "natural"  # Default style
+
+        # Set default size for models that need it
+        if model_name == "dall-e-3":
+            image_params["size"] = "1024x1024"
+        elif model_name == "gpt-image-1":
+            image_params["size"] = "1024x1024"
+        else:
+            image_params["size"] = "1024x1024"
+
+        try:
+            # gpt-image-1 doesn't support response_format parameter
+            if model_name == "gpt-image-1":
+                res = image_generation(**image_params)
+            else:
+                res = image_generation(response_format="url", **image_params)
+        except Exception as e:
+            raise
+
```

Comment:
I think these parameters should be sent in the request and validated by the client. We should throw an error if the params are not valid (can probably just propagate whatever the API sends back to the user